### PR TITLE
Fix users password reset by relaxing the username schema

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

Users are unable to change their passwords because their Usernames are not valid when using the default CKAN user schema, so relax the user schema by monkey patching it.

Also prevent organogram errors from reaching sentry as its an error that can be handled by the user and fix a bug in making the organogram filenames lowercase.

## Reference 

https://trello.com/c/alHbD02j/2705-users-cannot-change-their-ckan-password

### CODE REVIEW

File: ckanext/datagovuk/plugin.py
Comment (richardTowers) [inline ckanext/datagovuk/plugin.py:263]: It feels dangerous to have the monkey patching happen as a side effect of an import. Do you actually need `logic` and `query` here too?

Was there something wrong with the old approach? Or could we have a separate function on `ckan_patches` which does the monkey patching like:

```
from ckanext.datagovuk.ckan_patches.cli import monkey_patch
monkey_patch()
```

Comment (kentsanggds) [reply ckanext/datagovuk/plugin.py:263]: The old approach included these monkey patches. Logic is actually the one that handles the user schema, query blocks some search queries which were causing problems in the past. 
This has been fixed by upgrading the service affected but it's probably a good idea to continue blocking the tags.

Comment (richardTowers) [reply ckanext/datagovuk/plugin.py:263]: Before this PR, `logic` and `query` weren't being imported, or am I missing something? Are you saying that they used to be imported, before the upgrade, but they got removed at some point? So now we're adding them back?

I see the approach of monkey patching as a side effect of an import has been around for the last 5 years (in `logic.py`), so I guess I'm fine with it for this PR. It's a big code smell though - imports shouldn't have side effects, and they doubly shouldn't have monkey patching side effects.

File: ckanext/datagovuk/ckan_patches/cli.py
Comment (huwd) [inline ckanext/datagovuk/ckan_patches/cli.py:20]: What is the significance of this line moving from plugin.py to here in ckan_patches/cli.py?

File: ckanext/datagovuk/plugin.py
Comment (kentsanggds) [reply ckanext/datagovuk/plugin.py:263]: Pre 2.9 they were being imported, it was removed in this commit https://github.com/alphagov/ckanext-datagovuk/commit/270bc2da195cbc46ca38f2afcc0eec1d44da7647#diff-b5ec20d532632efd2db909f73caa5ff88a59ab0a1664dd6ceb3a8f4a31ae4df9 probably by accident as that commit was part of the 2.9 upgrade branch. 
I think the problem was that there weren't any tests in place to check that it was allowing through more relaxed user names as the user schema in DGU is much more relaxed, this test has now been added in to prevent this from happening in the future.

File: ckanext/datagovuk/upload.py
Comment (kentsanggds) [reply ckanext/datagovuk/upload.py:80]: Before we were using `awsome-slugify` which doesn't appear to work beyond python 3.4, this was only picked up after doing a `force-reinstall`, so I moved towards using `python-sulgify`, unfortunately `lowercase=True` is not the same as `to_lower=True` so I just used the standard `lower()` to ensure that the filename is lowercased.

File: ckanext/datagovuk/ckan_patches/cli.py
Comment (kentsanggds) [reply ckanext/datagovuk/ckan_patches/cli.py:20]: This keeps it more in line with the other patches in the `ckan_patches` directory, and also means that it can just be imported in `plugin.py`

**Review by richardTowers — APPROVED** [review]
Changes make sense now they've been explained.

I still think the pattern of monkey patching as a side effect of an import is extremely suspect, but it's been like that for many years now so it's not worth blocking this PR on that.

**Comment by huwd** [comment]
You can dismiss my questions and go with Richard's ✅  if you want,
I might still need insight into what's going on in those places before I feel informed enough to approve!

**Pull request merged**
